[CI] ShardSearchPhaseAPMMetricsTests testUniformCanMatchMetricAttributesWhenPlentyOfDocumentsInIndex failing

**Build Scans:**
- [elasticsearch-intake #40245 / part1](https://gradle-enterprise.elastic.co/s/74p3jo7joavfw)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.TelemetryMetrics.ShardSearchPhaseAPMMetricsTests.testUniformCanMatchMetricAttributesWhenPlentyOfDocumentsInIndex" -Dtests.seed=1BDB8F6BCEBDB615 -Dtests.locale=dyo-Latn-SN -Dtests.timezone=America/Barbados -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.TelemetryMetrics.ShardSearchPhaseAPMMetricsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testUniformCanMatchMetricAttributesWhenPlentyOfDocumentsInIndex'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size <5>
     but: collection size was <4>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testUniformCanMatchMetricAttributesWhenPlentyOfDocumentsInIndex
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -523,9 +523,6 @@ tests:
 - class: org.elasticsearch.indices.state.OpenCloseIndexIT
   method: testTranslogStats
   issue: https://github.com/elastic/elasticsearch/issues/149852
-- class: org.elasticsearch.search.TelemetryMetrics.ShardSearchPhaseAPMMetricsTests
-  method: testUniformCanMatchMetricAttributesWhenPlentyOfDocumentsInIndex
-  issue: https://github.com/elastic/elasticsearch/issues/149874
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {csv-spec:ints.valuesLongGrouped}
   issue: https://github.com/elastic/elasticsearch/issues/149905

--- a/server/src/test/java/org/elasticsearch/search/TelemetryMetrics/ShardSearchPhaseAPMMetricsTests.java
+++ b/server/src/test/java/org/elasticsearch/search/TelemetryMetrics/ShardSearchPhaseAPMMetricsTests.java
@@ -427,21 +427,23 @@ public class ShardSearchPhaseAPMMetricsTests extends ESSingleNodeTestCase {
         );
         ensureGreen(indexName);
 
+        long docs = randomIntBetween(50, 100);
         LocalDate baseDate = LocalDate.of(2024, 11, 1);
-        for (int i = 1; i <= num_primaries * 2; i++) {
+        var builder = client().prepareBulk(indexName);
+        for (int i = 1; i <= docs; i++) {
             LocalDate docDate = baseDate.plusMonths(i - 1);
-            prepareIndex(indexName).setId(Integer.toString(i))
-                .setSource("body", "doc" + i, "@timestamp", docDate.format(DateTimeFormatter.ISO_LOCAL_DATE))
-                .setRefreshPolicy(IMMEDIATE)
-                .get();
-
+            builder.add(
+                prepareIndex(indexName).setId(Integer.toString(i))
+                    .setSource("body", "doc" + i, "@timestamp", docDate.format(DateTimeFormatter.ISO_LOCAL_DATE))
+            );
         }
+        builder.setRefreshPolicy(IMMEDIATE).get();
 
         BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
         boolQueryBuilder.filter(new RangeQueryBuilder("@timestamp").from("2024-10-01"));
         assertResponse(client().prepareSearch(indexName).setPreFilterShardSize(1).setQuery(boolQueryBuilder), searchResponse -> {
             assertNoFailures(searchResponse);
-            assertHitCount(searchResponse, num_primaries * 2L);
+            assertHitCount(searchResponse, docs);
         });
         final List<Measurement> canMatchMeasurements = getTestTelemetryPlugin().getLongHistogramMeasurement(CAN_MATCH_SEARCH_PHASE_METRIC);
         assertEquals(num_primaries, canMatchMeasurements.size());

